File: Java/Training/src/kunai_test.java
import io.appium.java_client.AppiumDriver;
import io.appium.java_client.android.AndroidDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.remote.DesiredCapabilities;
import java.net.URL;
import java.util.concurrent.TimeUnit;

public class kunai_test
{
    public static void main( String[] args ) throws Exception
    {
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability( "appium-version", "1.0" );
        capabilities.setCapability( "platformName", "Android" );
        capabilities.setCapability( "platformVersion", "5.0.1" );
        capabilities.setCapability( "deviceName", "API21" );
        capabilities.setCapability( "appPackage", "com.cybozu.kunailite" );
        capabilities.setCapability( "appActivity", ".ui.MainActivity" );
        capabilities.setCapability( "autoLaunch", "false" );
        AppiumDriver wd = new AndroidDriver( new URL( "http://localhost:4723/wd/hub" ), capabilities );

        try
        {
            wd.launchApp();
            Thread.sleep( 3000 );
            wd.manage().timeouts().implicitlyWait( 60, TimeUnit.SECONDS );
            wd.findElement( By.xpath( "//*[@text='New']" ) ).click();
            wd.manage().timeouts().implicitlyWait( 60, TimeUnit.SECONDS );
            wd.findElement( By.id( "com.cybozu.kunailite:id/sc_schedule_title" ) ).sendKeys( new String[]{ "222" } );
            //wd.findElement( By.id( "com.cybozu.kunailite:id/sc_schedule_title" ) ).sendKeys( "111" );
            wd.findElement( By.xpath( "//*[@text='Add']" ) ).click();
            Thread.sleep( 3000 );

        }
        finally
        {
            wd.quit();
        }

    }
}

File: Python/Training/kunai_test.py
#coding=utf-8
'''
Created on 2016年1月20日

@author: QLLU
'''
from appium import webdriver
import time

desired_caps = {}
desired_caps['platformName'] = 'Android'
desired_caps['platformVersion'] = '5.0.2'
desired_caps['deviceName'] = 'emulator-5554'
desired_caps['appPackage'] = 'com.cybozu.kunailite'
desired_caps['appActivity'] = '.ui.MainActivity'
desired_caps['autoLaunch'] = 'false'
driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)

def open():
	if driver.is_app_installed('com.cybozu.kunailite') is True:
		driver.launch_app()
		print "run"
		time.sleep(3)
	else:
		print "Please install apk."
def close():
	driver.quit()

def test_1():
	try:
		open()
		# 翻页3次，向左滑屏3次
		for i in range(3):
			driver.swipe(400, 400, 10, 400, 300)
			# driver.flick(400, 400, 0, 400)
			time.sleep(1)
		time.sleep(5)
	except Exception as e:
		print e
	finally:
		close()

test_1()









